repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -122,7 +122,11 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
     }
 
     val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable match {
+      case x if x.nonEmpty => x
```

Comment:
You know what, just ignore this entire comment chain.

I cannot come up with a way of doing this that I really like. I kind of want a method like `getOrElse` on Option but for  collections. I want to treat an empty collection as being distinct from a non-empty collection.